src/building/warehouse.c

```diff
@@ -115,6 +115,28 @@ int building_warehouse_add_resource(building *b, int resource)
     return 1;
 }
 
+int building_warehouses_add_resource(int resource, int amount, int respect_settings)
+{
+    if (amount <= 0) {
+        return 0;
+    }
+    
+    int remaining_to_add = amount;
```

Comment:
Again, no need for the `remaining_to_add` variable. You can just use `amount`.

Reply:
Done.